pkg/tsdb/opentsdb/opentsdb.go

```diff
@@ -126,9 +126,8 @@ func (e *OpenTsdbExecutor) parseResponse(query OpenTsdbQuery, res *http.Response
 	}
 
 	for _, val := range data {
-		series := tsdb.TimeSeries{
-			Name: val.Metric,
-		}
+		series := tsdb.TimeSeries{}
+		series.Name = e.createSeriesLabel(query.Queries[0]["alias"].(string), val)
```

Comment:
yes, this looks very fragile

Reply:
I agree using query.Queries[0]["alias"] is indeed fragile (and, as per my understanding, incorrect as well).
I am struggling to determine which output time series maps to which input query object. Need some help here...

PS: A single query can return multiple time series in its output in case its a regex query and results from all input queries are not returned in a List<List<>> type structure by OpenTSDB, but are returned in a single List<> type structure (by using a flatMap() type operation.